### ISSUE TITLE
pigz-ng crc32 mismatch

### ISSUE THREAD

**Issue by abmybgx** [zlib-ng/zlib-ng#1565]
I built pigz-ng from pigz-bench last week, and I met below error. I tried WITH_OPTIM=OFF option, but it was no use. After I changed back to pigz, it works.

```
nerdctl pull marvinb/carla_simulator:0.9.14

cd /var/lib/containerd/root/io.containerd.content.v1.content/blobs/sha256

./pigz-ng -cd 918d710d636a791b817a550f8b6eb0b78290d90a813f93812d3b57b022ec7a55 > /dev/null
pigz-ng: skipping: 918d710d636a791b817a550f8b6eb0b78290d90a813f93812d3b57b022ec7a55: corrupted -- crc32 mismatch
```

**Comment by nmoinvaz** [comment]
Can you attach your input files that reproduce the problem? What architecture and platform are you using? Is there a particular zlib-ng commit that it works for? Also what version or commit of pigz are you using?

**Comment by KungFuJesus** [comment]
It also sounds like it could be an issue in pigz-ng, rather than zlib-ng.

**Comment by nmoinvaz** [comment]
@KungFuJesus there is no specific pigz-ng project.

**Comment by nmoinvaz** [comment]
The only thing that the pigz folder provides in pigzbench is a CMake project and Windows shim files.

**Comment by KungFuJesus** [comment]
> The only thing that the pigz folder provides in pigzbench is a CMake project and Windows shim files.

Hmm, the @abmybgx claims they are using "pigz-ng" and the failed test is specifically calling itself that.  It also sounds like when switching back to pigz with zlib-ng's compat mode, it had no issues.  Maybe I'm missing something, though.

**Comment by KungFuJesus** [comment]
Oh, the table on that project clarifies my confusion: https://github.com/zlib-ng/pigzbench

So it's just zlib-ng being tested with pigz.

**Comment by nmoinvaz** [comment]
There is a `a_compile.py` that compiles pigz with different variants of zlib. But I'm not sure if it was used to build the issue in this case.

**Comment by abmybgx** [comment]
My pigz-ng was built by a_compile.py with the newest zlib-ng develop branch, you all could follow my operation to have a test, it's a public image. For decompressed bytes file, there was 1 byte mismatch from pigz-ng to pigz. It's ok for pigz or igzip.

**Comment by nmoinvaz** [comment]
Ok. Maybe somebody else can attach the file, I don't use docker.

**Comment by mtl1979** [comment]
> Ok. Maybe somebody else can attach the file, I don't use docker.

That file is ~8.835 GB, so attaching the file is impossible...

**Comment by KungFuJesus** [comment]
@abmybgx does https://github.com/zlib-ng/zlib-ng/pull/1602 make this issue go away?

**Comment by KungFuJesus** [comment]
I decided to try to test this myself but got as far as looking at the content of the dockerfile to find that it's not doing anything that invokes whatever's going on here:

https://hub.docker.com/layers/marvinb/carla_simulator/0.9.14/images/sha256-a769bca47828a99f96dbb5093d3eeeddaa00fd84558d7cf7bf3d5adf913bb53e?context=explore

What is the actual command that you're running to produce this test failure?  I'd like to reproduce it without the mess of docker involved.

**Comment by nmoinvaz** [comment]
This should be resolved by the referenced commit and is available in 2.1.5 or higher.
